src/main/java/uk/gov/hmcts/reform/preapi/controllers/RecordingController.java

```diff
@@ -36,28 +44,49 @@ public RecordingController(RecordingService recordingService) {
     @GetMapping("/{recordingId}")
     @Operation(operationId = "getRecordingById", summary = "Get a Recording by Id")
     public ResponseEntity<RecordingDTO> getRecordingById(
-        @PathVariable UUID bookingId,
         @PathVariable UUID recordingId
     ) {
         // TODO Recordings returned need to be shared with the current user
-        return ResponseEntity.ok(recordingService.findById(bookingId, recordingId));
+        return ResponseEntity.ok(recordingService.findById(recordingId));
     }
 
     @GetMapping
-    @Operation(operationId = "getRecordingsByBookingId", summary = "Get all Recordings by Booking Id")
-    public ResponseEntity<List<RecordingDTO>> getAllRecordingsByBookingId(
-        @PathVariable UUID bookingId,
-        @RequestParam(required = false) UUID captureSessionId,
-        @RequestParam(required = false) UUID parentRecordingId
+    @Operation(operationId = "getRecordings", summary = "Search all Recordings")
+    @Parameter(
+        name = "captureSessionId",
+        description = "The capture session to search by",
+        example = "123e4567-e89b-12d3-a456-426614174000"
+    )
+    @Parameter(
+        name = "parentRecordingId",
+        description = "The parent recording to search by",
+        example = "123e4567-e89b-12d3-a456-426614174000"
+    )
+    public HttpEntity<PagedModel<EntityModel<RecordingDTO>>> searchRecordings(
+        @RequestParam Map<String, String> params,
+        Pageable pageable,
+        PagedResourcesAssembler<RecordingDTO> assembler
```

Comment:
There is a nasty quirk in the swagger docs I saw where it adds the assembler, pager and params in a way we don't want.
I don't know the solution so leave this as is for now but on Monday lets look at fixing this up as it makes creating the Component for Power Apps more long winded right now unfortunately as requires extra manual wrangling of the yaml prior to importing it.

Reply:
I think you can mark it as hidden... something like `@Parameter(hidden = true) Pageable pageable` will create another pr to fix that